### PR TITLE
add cmake options

### PR DESCRIPTION
**Pull request by bd4**
add GTENSOR_CHECK_BOUNDS, GTENSOR_CHECK_ADDRESS, GTENSOR_SYNC_KERNELS

### CODE REVIEW

**Comment by bd4** [comment]
I wonder if check address should be on by default - I don't expect to gtensor_span objects to be construted in a loop, and it aids in early detection of some type of errors.

File: CMakeLists.txt [hunk 1]
```diff
@@ -37,6 +37,10 @@ option(GTENSOR_PER_DIM_KERNELS
 
 option(GTENSOR_ALLOCATOR_CACHING "Enable naive caching allocators" ON)
 
+option(GTENSOR_BOUNDS_CHECK "Enable per access bounds checking" OFF)
+option(GTENSOR_ADDRESS_CHECK "Enable per access bounds checking" OFF)
```
Comment (germasch) [inline CMakeLists.txt:41]: A little copy&paste mistake here ;)

**Review by germasch — CHANGES_REQUESTED** [review]
Looks great, other than one tiny bit! ;)

File: CMakeLists.txt (hunk 1 above)
Comment (bd4) [reply CMakeLists.txt]: Good catch!

**Review by germasch — APPROVED** [review]

**Pull request merged**